File: Python/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
#         for i in range(len(nums)):
#             temp = target - nums[i]
#             if temp in nums and i != nums.index(temp):
#                 return [i, nums.index(temp)]

        dict_ = {}
        for i, j in enumerate(nums):
            temp = target - j
            if temp in dict_:
                return [dict_[temp], i]
            else: dict_[j] = i

File: PHP/1. Two Sum.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $arr = array();

        for ($i = 0; $i < count($nums); $i++) {
            if (in_array($target - $nums[$i], $arr))
                return array($i, array_search($target - $nums[$i], $arr));
            
            $arr[$i] = $nums[$i];
        }

        return array(0, 0);
    }
}
